### PR TITLE
schedulers: skip uniform store to reduce schedule

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/4949

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by HunDunDM — APPROVED** [review]

File: server/schedulers/hot_region.go [hunk 1]
```diff
@@ -515,11 +533,9 @@ func (bs *balanceSolver) solve() []*operator.Operator {
...
 
 	bs.cur = &solution{}
-	tryUpdateBestSolution := func(isUniformFirstPriority bool) {
-		if bs.cur.progressiveRank == -1 && isUniformFirstPriority {
-			// Because region is available for src and dst, so stddev is the same for both, only need to calcurate one.
-			// If first priority dim is enough uniform, -1 is unnecessary and maybe lead to worse balance for second priority dim
-			hotSchedulerResultCounter.WithLabelValues("skip-uniform-store", strconv.FormatUint(bs.cur.dstStore.GetID(), 10)).Inc()
+	tryUpdateBestSolution := func() {
+		if label, ok := bs.filterUniformStore(); ok {
+			schedulerCounter.WithLabelValues(bs.sche.GetName(), fmt.Sprintf("%s-skip-%s-uniform-store", bs.rwTy.String(), label)).Inc()
```
Comment (CabinfeverB) [inline server/schedulers/hot_region.go:538]: Why not still using hotSchedulerResultCounter

**Review by CabinfeverB — APPROVED** [review]
rest LGTM

File: server/schedulers/hot_region.go (hunk 1 above)
Comment (lhy1024) [reply server/schedulers/hot_region.go:538]: The previous monitoring location is wrong

**Pull request merged**